File: Top-Interview-Questions-Leetcode/Easy-Collections/strings/valid-anagram.cpp
#include <iostream>
#include <vector>
#include <string>

class Solution {
public:
    bool isAnagram(std::string s, std::string t) {
        std::vector<int> cnt(26);
        
        for (char ch : s) cnt[ch - 'a']++;
        
        for (char ch : t) cnt[ch - 'a']--;
        
        for (int val : cnt) if (val) return false;
        
        return true;
    }
};

// Example usage
int main() {
    Solution solution;
    std::string s = "phanthanhdang";
    std::string t = "dangphanthanh";
    
    if (solution.isAnagram(s, t)) {
        std::cout << s << " and " << t << " are anagrams." << std::endl;
    } else {
        std::cout << s << " and " << t << " are not anagrams." << std::endl;
    }
    
    return 0;
}

File: Top-Interview-Questions-Leetcode/Easy-Collections/strings/valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        cnt = [0 for _ in range(26)]
        
        for ch in s:
            cnt[ord(ch) - ord('a')] += 1
        
        for ch in t:
            cnt[ord(ch) - ord('a')] -= 1
            
        for val in cnt:
            if val != 0:
                return False
        
        return True

# Example usage
def main():
    s = "dangphanthanh"
    t = "thanhdangphan"
    solution = Solution()
    result = solution.isAnagram(s, t)
    print(f"Is '{s}' an anagram of '{t}'? {result}")

if __name__ == "__main__":
    main()